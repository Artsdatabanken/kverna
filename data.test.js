const { io, log } = require("lastejobb")
const config = require("./config")
var { tellBarnasNøkler } = require("./lib/testHelper")

const dataFiles = io.findFiles(config.dataRoot, ".json")
const buildFiles = io.findFiles(config.buildRoot, ".json")
const files = dataFiles.concat(buildFiles)

files.forEach(file => {
  log.warn(file)
  const json = io.readJson(file)
  describe(file, () => {
    it("har forventede nøkler", () => {
      const actual = tellBarnasNøkler(json)
      expect(actual).toMatchSnapshot()
    })
    it("har antall rader", () => {
      const actual = Object.keys(json).length
      expect(actual).toMatchSnapshot()
    })
  })
})
