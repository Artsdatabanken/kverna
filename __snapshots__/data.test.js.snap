// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data/1_kildedata/MI_variasjon.json har antall rader 1`] = `2909`;

exports[`data/1_kildedata/MI_variasjon.json har forventede nøkler 1`] = `
Object {
  "Kode": 2909,
  "Navn": 2909,
  "OverordnetKode": 2909,
}
`;

exports[`data/1_kildedata/NA_allekoder.json har antall rader 1`] = `1796`;

exports[`data/1_kildedata/NA_allekoder.json har forventede nøkler 1`] = `
Object {
  "ElementKode": 1696,
  "Kategori": 1796,
  "Kode": 1796,
  "Navn": 1796,
  "OverordnetKode": 1796,
}
`;

exports[`data/2_konvertering/MI_variasjon.json har antall rader 1`] = `2909`;

exports[`data/2_konvertering/MI_variasjon.json har forventede nøkler 1`] = `
Object {
  "foreldre": 2909,
  "kode": 2909,
  "tittel": 2909,
}
`;

exports[`data/2_konvertering/NA_hovedtyper.json har antall rader 1`] = `92`;

exports[`data/2_konvertering/NA_hovedtyper.json har forventede nøkler 1`] = `
Object {
  "definisjonsgrunnlag": 92,
  "kunnskap": 92,
  "lkm": 92,
  "nin1": 92,
  "nivå": 92,
  "prosedyrekategori": 92,
}
`;

exports[`data/2_konvertering/NA_koder.json har antall rader 1`] = `1796`;

exports[`data/2_konvertering/NA_koder.json har forventede nøkler 1`] = `
Object {
  "foreldre": 1796,
  "tittel": 1796,
}
`;

exports[`data/2_konvertering/NA_overstyr_hierarki.json har antall rader 1`] = `851`;

exports[`data/2_konvertering/NA_overstyr_hierarki.json har forventede nøkler 1`] = `
Object {
  "0": 851,
}
`;
